### PR TITLE
Add docs page about experimental features

### LINKED ISSUE #3920: Experimental features need to be properly documented
We have a few experimental features that need to be clearly marked as such, in a proper doc.

One example is Pluggables, which is still marked as experimental. Do we graduate it? If so, we should clearly document its use case and how it's supposed to be used internally in Volto.

### PR DESCRIPTION
**Pull request by davisagli**
Fixes #3920

### CODE REVIEW

File: docs/source/configuration/experimental.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+myst:
+  html_meta:
+    "description": "Enable experimental features in Volto's configuration object"
+    "property=og:description": "Enable experimental features in Volto's configuration object"
+    "property=og:title": "Experimental features"
+    "keywords": "Volto, Plone, frontend, experimental"
```
Comment (stevepiercy) [inline docs/source/configuration/experimental.md:7]:
```suggestion
    "keywords": "Volto, Plone, frontend, experimental, features"
```

File: docs/source/configuration/experimental.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+# Experimental features
```
Comment (stevepiercy) [inline docs/source/configuration/experimental.md:10]: Is there a relation between "experimental features" and a "canary" release? Thinking out loud here... For example, does a canary release include experimental features? I happened to mention that in another PR:

https://github.com/plone/volto/pull/3934/files#diff-c060614717ad6cda93b44a2e34baa9d862e93884d0318798631e34808ff3842dR11-R13

Comment (stevepiercy) [reply docs/source/configuration/experimental.md:10]: Oh, also "feature flags". Does that belong in here?

**Review by stevepiercy — CHANGES_REQUESTED** [review]
LGTM with one more keyword and a couple of questions.

File: docs/source/configuration/experimental.md (hunk 1 above)
Comment (davisagli) [reply docs/source/configuration/experimental.md:7]: @stevepiercy Sure, we can add this. Is it generally necessary to put words in keywords that are already in the title?

Comment (stevepiercy) [reply docs/source/configuration/experimental.md:7]: They are converted to meta tags. Just standard SEO stuff.

File: docs/source/configuration/experimental.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Experimental features are planned for inclusion in a future release of Volto,
+but are not yet considered mature by the community.
+
```
Comment (stevepiercy) [inline docs/source/configuration/experimental.md:14]:
```suggestion
An experimental feature would typically be released in a canary release, giving users an easy way to install it and see if anything breaks before we make a new stable release that includes it.

```

File: docs/source/configuration/experimental.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+## Volto configuration
+
+You need to enable experimental features in Volto's configuration object:
```
Comment (stevepiercy) [inline docs/source/configuration/experimental.md:17]:
```suggestion
You can enable an experimental feature—also called a "feature flag"—in Volto's configuration object as shown below.
```

**Review by stevepiercy — COMMENTED** [review]
@davisagli I wordsmithed your comments into a couple of suggestions.

Either way, this is good to go.

File: docs/source/configuration/experimental.md (hunk 3 above)
Comment (davisagli) [reply docs/source/configuration/experimental.md:14]: @stevepiercy I don't think this really helps clarify. That's true of non-experimental features too.

Comment (stevepiercy) [reply docs/source/configuration/experimental.md:14]: Maybe this?

```suggestion
An experimental feature gives users an easy way to install it and see if anything breaks before we make a new stable release that includes it.

```

**Review by stevepiercy — APPROVED** [review]

**Review by sneridagh — APPROVED** [review]

**Pull request merged**